### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by GoddessNgozi**
## In this PR I implemented the following:

- Use fetch with createAsyncThunk) for making API calls
- Fetch book list from Bookstore API on Books page load.
- Add an Action Creator, that returns a function.
- Fetch list of books.
- Dispatch action that passes API response to reducer and updates the state.
- Persist state changes in the server.
- Add book and remove book from server

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,48 +1,60 @@
...
 // Reducer
 const booksReducer = (state = defaultState, action) => {
   switch (action.type) {
+    case GET_BOOKS:
+      return action.payload;
     case ADD_BOOK:
       return [...state, action.payload];
     case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.payload.id);
+      return [...state.filter((book) => book.id !== action.payload.id)];
```
Comment (HAMMAS-SALEEM) [inline src/redux/books/books.js]: - [ ] When i try to remove your book, your app successfully removes the book from the API 😊 But the problem is after clicking on the remove button i always need to refresh the page to remove that book from the DOM. Kindly fix this issue 👍 to make your application respond like a single page application 🤔

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @GoddessNgozi :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following GitFlow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (GoddessNgozi) [reply src/redux/books/books.js:18]: @HAMMAS-SALEEM please can you hint how to remove it from the DOM too? Thank you

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

## Hey @GoddessNgozi :wave:

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:


## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following GitFlow
✔️ Following requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**